### DISCUSSION TITLE
Delete saved SSID in captive portal mode

### DISCUSSION THREAD

**Discussion by usystem** [Hieromon/AutoConnect#433] [Feature Request]
Hi! I only use the captive portal mode. In this mode, there is no way to delete the saved SSID (or I just don't know?). I suggest in the menu AC_MENUITEM_OPENSSIDS for each saved SSID to make a deletion button.

**Comment by Hieromon** [comment]
I don't think it's that important to include the interactive delete function in the library standard. You can use [Credentail.ino](https://github.com/Hieromon/AutoConnect/blob/master/examples/Credential/Credential.ino) as a reference to incorporate your desired extensions into your own sketches as needed.

However, if there are a lot of requests from other users, I would be willing to consider implementing a delete function. See what the response to this topic is.

**Reply by zealtv** [reply]
+1 for the ability to interactively delete SSIDs.  This would be extremely useful in several of my projects.   Thanks for your work on this 🍻

**Reply by Hieromon** [reply]
Planned to release as v1.3.4.
The followings are candidate specifications. Accept requests for specification changes by reply.

- Add a trash icon to the SSID list on the AC_MENUITEM_OPENSSIDS page to remove the credential.
- The trash icon is hidden by default.
- To enable the delete icon, you need to configure it with [AutoConnectConfig::menuItems](https://hieromon.github.io/AutoConnect/menu.html#applying-the-active-menu-items), `AC_MENUITEM_t::AC_MENUITEM_DELETESSID` enumeration value is newly added.
  ```cpp
  AutoConnect portal;
  AutoConnectConfig config;
  config.menuItems = config.menuItems | AC_MENUITEM_DELETESSID;
  portal.config(config);
  ```
- Prompts a dialog to confirm for deletion by clicking on the trash icon.
- The result of the deletion will be responded to in the AC_MENUITEM_OPENSSIDS page. (reloading a page)
- Authentication with [AutoConnectConfig::auth](https://hieromon.github.io/AutoConnect/adauthentication.html#applying-http-authentication) is available, but only the delete icon cannot be subject to authentication. The entire page will be subject to authentication.

![openssid_trash](https://user-images.githubusercontent.com/12591771/153234926-57b2acf8-26e0-4872-8bac-d184a51c301e.png)

**Reply by zealtv** [reply]
This looks very well thought out!  I'm keeping a close eye and will test the first chance 🍔

**Comment by usystem** [comment]
Great news! We are looking forward to it!

**Comment by Hieromon** [comment]
The ability to delete credentials in the Open SSID menu has been released as [AutoConnect v1.3.4](https://github.com/Hieromon/AutoConnect/releases/tag/v1.3.4).

**Reply by usystem** [reply]
Good job! Thanks a lot! All perfectly work.

**Comment by klkl140** [comment]
Works great. Now I can delete my own routine for deleting all stored connections. Thank you
